src/sources/apache_metrics/mod.rs

```diff
@@ -0,0 +1,394 @@
+
+                            for err in errors {
+                                emit!(ApacheMetricsParseError {
+                                    error: err.into(),
+                                    url: url.clone(),
+                                });
+                            }
+
+                            metrics.push(Metric {
+                                name: "apache_up".into(),
+                                timestamp: Some(Utc::now()),
+                                tags: Some(tags.clone()),
+                                kind: MetricKind::Absolute,
+                                value: MetricValue::Gauge { value: 0.0 },
+                            });
+
+                            emit!(ApacheMetricsEventReceived {
+                                byte_size,
+                                count: metrics.len(),
+                            });
+                            Some(stream::iter(metrics).map(Event::Metric).map(Ok))
+                        }
+                        Ok((header, _)) => {
+                            emit!(ApacheMetricsErrorResponse {
+                                code: header.status,
+                                url: url.clone(),
+                            });
+                            Some(
+                                stream::iter(vec![Metric {
+                                    name: "apache_up".into(),
+                                    timestamp: Some(Utc::now()),
+                                    tags: Some(tags.clone()),
+                                    kind: MetricKind::Absolute,
+                                    value: MetricValue::Gauge { value: 1.0 },
+                                }])
+                                .map(Event::Metric)
+                                .map(Ok),
+                            )
+                        }
+                        Err(error) => {
+                            emit!(ApacheMetricsHttpError {
+                                error,
+                                url: url.clone()
+                            });
+                            Some(
+                                stream::iter(vec![Metric {
+                                    name: "apache_up".into(),
+                                    timestamp: Some(Utc::now()),
+                                    tags: Some(tags.clone()),
+                                    kind: MetricKind::Absolute,
+                                    value: MetricValue::Gauge { value: 0.0 },
+                                }])
+                                .map(Event::Metric)
+                                .map(Ok),
+                            )
+                        }
+                    })
+                })
+                .flatten()
+        })
+        .flatten()
+        .forward(out)
+        .inspect(|_| info!("finished sending"));
+
+    Box::new(task.boxed().compat())
+}
+
+#[cfg(feature = "sources-apache_metrics")]
```

Comment:
I don't believe this is needed, since the whole module is behind the same feature flag.